### PR TITLE
Fix nextPowOf2 on zero input

### PR DESCRIPTION
**Pull request by rfyiamcool**
### change

1. if input `0` in nextPowOf2, the v is max(uint32) - 1 after v--.
2. add more unit test for util.

### CODE REVIEW

File: util.go [hunk 1]
```diff
@@ -20,6 +20,9 @@ const (
 // nextPowOf2 computes the next highest power of 2 of 32-bit v.
 // Source: https://graphics.stanford.edu/~seander/bithacks.html#RoundUpPowerOf2
 func nextPowOf2(v uint32) uint32 {
+	if v == 0 {
+		return 2
```
Comment (puzpuzpuz) [inline util.go:24]: Is there a reason not to use 1 as the return value for 0 input instead of 2? After all `nextPowOf2(1) == 1`.

**Review by puzpuzpuz — COMMENTED** [review]
Thanks for the contribution! Left a question on the `nextPowOf2(0)` value.

File: util.go (hunk 1 above)
Comment (rfyiamcool) [reply util.go]: 😁 done, `nextPowOf2(1) == 1`

**Review by puzpuzpuz — APPROVED** [review]
Thanks for the fix!

**Pull request merged**